test/functional/find-options/basic-usage/find-options-relations.ts

```diff
@@ -45,9 +44,6 @@ describe("find options > relations", () => {
 
     it("complex relation #1", () => Promise.all(connections.map(async connection => {
```

Comment:
can you please tell why previous version didn't work and new version works?

Reply:
The current ordering functionality is fundamentally broken. The reasoning is covered in a comment further up in this PR. This is an _attempt_ to work around that, however realistically the underlying functionality needs to be fixed.

However, a public facing API change is required, so I'm reluctant to make such a change myself.